ucp/core.py

```diff
@@ -472,28 +478,17 @@ class Endpoint:
     to create an Endpoint.
     """
 
-    def __init__(
-        self,
-        endpoint,
-        ctx,
-        msg_tag_send,
-        msg_tag_recv,
-        ctrl_tag_send,
-        ctrl_tag_recv,
-        guarantee_msg_order,
-    ):
+    def __init__(self, endpoint, ctx, guarantee_msg_order, tags=None):
         self._ep = endpoint
         self._ctx = ctx
-        self._msg_tag_send = msg_tag_send
-        self._msg_tag_recv = msg_tag_recv
-        self._ctrl_tag_send = ctrl_tag_send
-        self._ctrl_tag_recv = ctrl_tag_recv
         self._guarantee_msg_order = guarantee_msg_order
         self._send_count = 0  # Number of calls to self.send()
         self._recv_count = 0  # Number of calls to self.recv()
         self._finished_recv_count = 0  # Number of returned (finished) self.recv() calls
         self._shutting_down_peer = False  # Told peer to shutdown
         self._close_after_n_recv = None
+        self._tags = tags
+        assert tags is not None
```

Comment:
I don't think this line makes sense. Why have `tags=None` in the kwargs and assert here it isn't `None`?

Reply:
Oops, you are right that was only for debugging :)